### PR TITLE
Update `@typescript-eslint` to v8

### PR DESCRIPTION
**Pull request by amanmahajan7**
https://github.com/typescript-eslint/typescript-eslint/releases/tag/v8.0.0
https://typescript-eslint.io/blog/announcing-typescript-eslint-v8/
https://github.com/typescript-eslint/typescript-eslint/discussions/8914
https://github.com/jest-community/eslint-plugin-jest/blob/main/CHANGELOG.md

### CODE REVIEW

File: eslint.config.js [hunk 1]
```diff
@@ -202,6 +196,7 @@ export default [
       'no-restricted-imports': 0, // replaced by @typescript-eslint/no-restricted-imports
       'no-restricted-properties': 0,
       'no-restricted-syntax': 0,
+      'no-restricted-types': 0,
```
Comment (amanmahajan7) [inline eslint.config.js:199]: https://typescript-eslint.io/rules/no-restricted-types/

File: eslint.config.js [hunk 2]
```diff
@@ -422,10 +417,11 @@ export default [
       '@typescript-eslint/await-thenable': 1,
       '@typescript-eslint/ban-ts-comment': [1, { 'ts-expect-error': false }],
       '@typescript-eslint/ban-tslint-comment': 0,
-      '@typescript-eslint/ban-types': 1,
       '@typescript-eslint/class-literal-property-style': 0,
+      '@typescript-eslint/class-methods-use-this': 1,
```
Comment (amanmahajan7) [inline eslint.config.js:421]: https://typescript-eslint.io/rules/class-methods-use-this/

File: eslint.config.js [hunk 3]
```diff
@@ -422,10 +417,11 @@ export default [
...
       '@typescript-eslint/consistent-generic-constructors': 1,
       '@typescript-eslint/consistent-indexed-object-style': 1,
+      '@typescript-eslint/consistent-return': 0,
```
Comment (amanmahajan7) [inline eslint.config.js:424]: https://typescript-eslint.io/rules/consistent-return/ This seems too strict

File: eslint.config.js [hunk 4]
```diff
@@ -446,7 +442,6 @@ export default [
       '@typescript-eslint/no-duplicate-enum-values': 1,
       '@typescript-eslint/no-duplicate-type-constituents': 1,
       '@typescript-eslint/no-dynamic-delete': 0,
-      '@typescript-eslint/no-empty-interface': 1,
```
Comment (amanmahajan7) [inline eslint.config.js:449]: Deprecated
https://typescript-eslint.io/rules/no-empty-interface/

File: eslint.config.js [hunk 5]
```diff
@@ -481,13 +476,17 @@ export default [
       '@typescript-eslint/no-unsafe-call': 0,
       '@typescript-eslint/no-unsafe-declaration-merging': 1,
       '@typescript-eslint/no-unsafe-enum-comparison': 1,
+      '@typescript-eslint/no-unsafe-function-type': 1,
```
Comment (amanmahajan7) [inline eslint.config.js:479]: https://typescript-eslint.io/rules/no-unsafe-function-type/

File: eslint.config.js [hunk 6]
```diff
@@ -481,13 +476,17 @@ export default [
...
       '@typescript-eslint/no-unsafe-member-access': 0,
       '@typescript-eslint/no-unsafe-return': 1,
+      '@typescript-eslint/no-unsafe-unary-minus': 1,
```
Comment (amanmahajan7) [inline eslint.config.js:482]: https://typescript-eslint.io/rules/no-unsafe-unary-minus/

File: eslint.config.js [hunk 7]
```diff
@@ -481,13 +476,17 @@ export default [
...
       '@typescript-eslint/no-useless-empty-export': 1,
       '@typescript-eslint/no-var-requires': 0,
+      '@typescript-eslint/no-wrapper-object-types': 1,
```
Comment (amanmahajan7) [inline eslint.config.js:485]: https://typescript-eslint.io/rules/no-wrapper-object-types/

File: eslint.config.js [hunk 8]
```diff
@@ -481,13 +476,17 @@ export default [
...
       '@typescript-eslint/non-nullable-type-assertion-style': 1,
       '@typescript-eslint/parameter-properties': 1,
       '@typescript-eslint/prefer-as-const': 1,
+      '@typescript-eslint/prefer-destructuring': [1, { array: false }],
```
Comment (amanmahajan7) [inline eslint.config.js:489]: https://typescript-eslint.io/rules/prefer-destructuring/

File: eslint.config.js [hunk 9]
```diff
@@ -527,13 +526,13 @@ export default [
       '@typescript-eslint/default-param-last': 0,
       '@typescript-eslint/dot-notation': 1,
       '@typescript-eslint/init-declarations': 0,
+      '@typescript-eslint/max-params': 0,
```
Comment (amanmahajan7) [inline eslint.config.js:529]: https://typescript-eslint.io/rules/max-params/

File: eslint.config.js [hunk 10]
```diff
@@ -527,13 +526,13 @@ export default [
...
       '@typescript-eslint/no-array-constructor': 1,
       '@typescript-eslint/no-dupe-class-members': 0,
       '@typescript-eslint/no-empty-function': 0,
       '@typescript-eslint/no-implied-eval': 1,
       '@typescript-eslint/no-invalid-this': 0,
       '@typescript-eslint/no-loop-func': 0,
-      '@typescript-eslint/no-loss-of-precision': 1,
```
Comment (amanmahajan7) [inline eslint.config.js:536]: https://typescript-eslint.io/rules/no-loss-of-precision/

File: eslint.config.js [hunk 11]
```diff
@@ -571,20 +570,6 @@ export default [
       '@typescript-eslint/no-use-before-define': 0,
       '@typescript-eslint/no-useless-constructor': 1,
       '@typescript-eslint/only-throw-error': 1,
-      // TODO: deprecated
```
Comment (amanmahajan7) [inline eslint.config.js:574]: This was removed

File: eslint.config.js (hunk 1 above)
Comment (amanmahajan7) [reply eslint.config.js:199]: You are right

**Review by nstepien — APPROVED** [review]

**Pull request merged**